Documentation/network/standalone.rst

```diff
@@ -0,0 +1,1610 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+.. _standalone-gateway:
+
+*************************
+Cilium Standalone Gateway
+*************************
+
+Cilium has a standalone mode where it can be operated outside of Kubernetes
+as a standalone gateway. The gateway is designed to reside at the edge of the
+cluster, potentially replacing dedicated hardware components, and handling
+high-traffic load from a north-south direction going into the cluster(s).
+
+The Cilium Standalone Gateway includes functionality such as a high performance
+eBPF-based layer 4 load-balancer and a NAT46/64 gateway. There is a wide range
+of use cases, including:
+
+- Frontend DNS resolver
+- Load-balancer for legacy VM-based workloads,
+- As a gateway in front of IPv6 single stack Kubernetes clusters to allow for
+  interconnection with IPv4,
+- As a load-balancer for kubeapi-server,
+- As a replacement for legacy IPVS-based load-balancers,
+- And many more!
+
+Cilium provides a programmable API for its agent which is exposed through a
+unix domain socket on the local node such that it can be integrated into
+third party management software on the node.
+
+This guide explains how to provision Cilium as a standalone gateway along with
+various use-cases and configuration examples.
+
+Quick-Start
+===========
+
+Given the Cilium Standalone Gateway can be operated outside of a Kubernetes
+cluster, this guide provides minimal Docker-based step-by-step instructions.
+
+As a one-time prerequisite, ensure that the BPF file system is mounted under
+``/sys/fs/bpf``. If this is not the case already, a mount unit file can be set
+up under ``/etc/systemd/system/sys-fs-bpf.mount`` ...
+
+.. parsed-literal::
+
+   [Unit]
+   Description=BPF mounts
+   DefaultDependencies=no
+   Before=local-fs.target umount.target
+   After=swap.target
+
+   [Mount]
+   What=bpffs
+   Where=/sys/fs/bpf
+   Type=bpf
+
+   [Install]
+   WantedBy=multi-user.target
+
+... and brought up through ``systemctl start sys-fs-bpf.mount``.
+
+.. note::
+
+  The above step is only required for distributions with older systemd versions.
+  For most modern distributions this step is not required as systemd will
+  automatically mount the BPF file system at its default location. See also
+  `GH-10955 <https://github.com/cilium/cilium/issues/10955>`_ for details.
```

Comment:
I wonder if we should just put this step in some other requirements file that you have to click through, then simplify this doc to just say "Ensure that the BPF filesystem is mounted. This is enabled by default in all modern distributions, but if you are unsure then see [reference to the other page]".

This would allow a bunch of configuration output here to be hidden which seems ideal given that 90% of users shouldn't need it.